move to python pod serving up ignition instead of S3

joelddiaz opened the pull request:
Rather than set up an S3 bucket to provide the ignition config to the worker nodes, set up a Pod on the hypershift host to serve up ignition.

Remove the S3 bucket creation/deletion code as it is no longer needed.

Commit:
vendor new dependencies

Commit:
move to python pod serving up ignition instead of S3

Rather than set up an S3 bucket to provide the ignition config to the worker nodes, set up a Pod on the hypershift host to serve up ignition.

Remove the S3 bucket creation/deletion code as it is no longer needed.

csrwng commented on pkg/aws/install.go:
```diff
@@ -537,6 +553,162 @@ func InstallCluster(name, releaseImage, dhParamsFile string, waitForReady bool)
+		Port: &routev1.RoutePort{
+			TargetPort: intstr.IntOrString{
+				Type:   intstr.String,
+				StrVal: "https",
```
s/https/http/

joelddiaz replied:
i'm just going to remove the ports as it will just use everything (all one ports) defined in the Service

csrwng commented on pkg/aws/install.go:
```diff
@@ -537,6 +553,162 @@ func InstallCluster(name, releaseImage, dhParamsFile string, waitForReady bool)
+		Type: corev1.ServiceTypeClusterIP,
+		Ports: []corev1.ServicePort{
+			{
+				Name:     "https",
```
s/https/http/

csrwng commented on pkg/aws/install.go:
```diff
@@ -537,6 +553,162 @@ func InstallCluster(name, releaseImage, dhParamsFile string, waitForReady bool)
+			{
+				Name:     "https",
+				Protocol: corev1.ProtocolTCP,
+				Port:     443,
```
I would leave this as 8000 or 80... given that the target is not TLS, just plain-text http

csrwng commented on pkg/aws/install.go:
```diff
@@ -537,6 +553,162 @@ func InstallCluster(name, releaseImage, dhParamsFile string, waitForReady bool)
+				StrVal: "https",
+			},
+		},
+		TLS: &routev1.TLSConfig{
```
I would remove the TLS bit unless you plan to include the CA in the ignition file. This should be a plain-text http for now. Later on we'll need to think harder about security.

csrwng reviewed (commented):
@joelddiaz this is awesome. Some comments about the service/route.
For the future we need to think how to secure this payload given that it can be used to add arbitrary workers to the cluster.
Also (for a follow-up) we may want to stop generating manifests in code and follow the pattern in hypershift-toolkit where we have templated manifests as assets and just render those. It will make things easier to maintain going forward.

joelddiaz commented:
@csrwng i moved to yaml files that we can more easily update, and just sore them with the go-bindata tool. Also fixed up the https/tls bits that you suggested. PTAL, and i'll squash

csrwng commented:
lgtm
ship it!

joelddiaz commented:
sqashed, so merging
